src/components/layout/header.tsx

```diff
@@ -1,82 +1,98 @@
 import hoverDropdownIcon from "../../assets/hover-dropdown-icon.svg";
-import { useRef } from 'react';
-import {links} from "../../utils/links";
+import { useRef } from "react";
+import type { NavigationSection } from "../../utils/api";
 
 const LazyHeaderExpanded = React.lazy(() => import("./header-expanded"));
 
-// const headerSections: string[] = [
-//     "Products", "Learning", "Docs", "Models", "News", "Community", "About"
-// ];
-
-let headerSections: string[] = Object.keys(links.Header);
-
+interface HeaderProps {
+  navigation_sections: NavigationSection[];
+}
 
 interface HeaderActionProps {
-    title: string;
-    isHovered: boolean;
-    onMouseEnter: () => void;
-    isCompact: boolean
+  title: string;
+  isHovered: boolean;
+  onMouseEnter: () => void;
+  isCompact: boolean;
 }
 
-const HeaderAction = React.memo(({ title, isHovered, onMouseEnter, isCompact }: HeaderActionProps) => (
-    <div 
-        className="header-action-cont"
-        onMouseEnter={onMouseEnter}
-    >
-        <div className={`header-action ${isHovered ? 'header-action-hovered' : ''}`}>{title}</div>
-        <img className="dropdown-icon" src={isHovered ? hoverDropdownIcon.src : dropdownIcon.src} alt="dropdown"/>
+const HeaderAction = React.memo(
+  ({ title, isHovered, onMouseEnter, isCompact }: HeaderActionProps) => (
+    <div className="header-action-cont" onMouseEnter={onMouseEnter}>
+      <div
+        className={`header-action ${isHovered ? "header-action-hovered" : ""}`}
+      >
+        {title}
+      </div>
+      <img
+        className="dropdown-icon"
+        src={isHovered ? hoverDropdownIcon.src : dropdownIcon.src}
+        alt="dropdown"
+      />
     </div>
-));
+  )
+);
 
-const Header = () => {
-    const headerRef = useRef(null);
-    const [isCompact, setIsCompact] = useState(false);
-    const [hoveredIndex, setHoveredIndex] = useState<number | null>(null);
-    const handleMouseEnter = useCallback((index: number) => {
-        setHoveredIndex(index);
-    }, []);
+const Header = ({ navigation_sections }: HeaderProps) => {
+  const headerRef = useRef(null);
+  const [isCompact, setIsCompact] = useState(false);
+  const [hoveredIndex, setHoveredIndex] = useState<number>(-1);
+  const handleMouseEnter = useCallback((index: number) => {
+    setHoveredIndex(index);
+  }, []);
 
-    const handleHeaderMouseLeave = useCallback(() => {
-        setHoveredIndex(-1);
-    }, []);
+  const handleHeaderMouseLeave = useCallback(() => {
+    setHoveredIndex(-1);
+  }, []);
 
-    const memoizedHeaderActions = useMemo(() => 
-        headerSections.map((title, index) => (
-            <HeaderAction 
-                key={index} 
-                title={title} 
-                isHovered={hoveredIndex === index}
-                onMouseEnter={() => handleMouseEnter(index)}
-                isCompact={isCompact}
-            />
-        ))
-    , [hoveredIndex, handleMouseEnter]);
+  const memoizedHeaderActions = useMemo(
```

Comment:
Why should the header actions be memoized? I don't know that much about React, but from what I understand, memoizing is to skip re-rendering for performance considerations. But we shouldn't ever be re-rendering since it is a static site, right?